### PR TITLE
OIDC Hmac pairwise mapper for email

### PR DESCRIPTION
**Pull request by ingokuba**

### CODE REVIEW

**Comment by jens-hammel** [comment]
I miss the config to not overwrite the email if it was already set.

File: hmac-mapper/src/main/java/de/intension/protocol/oidc/mappers/HmacPairwiseEmailMapper.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        return "HMAC pairwise email";
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Updates the email claim in tokens with a pseudonymized value from a selected attribute.";
+    }
+
+    @Override
+    public IDToken transformIDToken(IDToken token, ProtocolMapperModel mappingModel, KeycloakSession session,
```
Comment (jens-hammel) [inline hmac-mapper/src/main/java/de/intension/protocol/oidc/mappers/HmacPairwiseEmailMapper.java:49]: I thought this should only be done if it is not set. Ticket says it should, and not must but ...

Comment (ingokuba) [reply hmac-mapper/src/main/java/de/intension/protocol/oidc/mappers/HmacPairwiseEmailMapper.java:49]: Added configuration for this - forgot about the OIDCAttributeMapperHelper

**Review by jens-hammel — DISMISSED** [review]
I approve but think we should wait with merge since deployment then starts and I not yet trust the rollingUpdatePolicy.

**Pull request merged**